astropy/units/si.py

```diff
@@ -29,12 +29,22 @@
 ###########################################################################
 # LENGTH
 
+# We exclude c as a prefix so we can define cm as a derived unit,
+# not a PrefixUnit. That way it can be used in cgs as a base unit.
```

Comment:
This will change types. I think this is an excellent indication that we need a base class for all units so that `isinstance(unit, UnitBaseClass)` always works.

Reply:
That is a bit a separate discussion - more relevant for magnitudes, etc. As is,
```
In [2]: isinstance(u.mm, u.UnitBase)
Out[2]: True
```
I think it is more that `PrefixUnit` is explicitly filtered out in `.compose()`.